api/jobs/queue.py

```diff
@@ -417,13 +417,26 @@ def scan_for_orphans():
         """
 
         orphaned = 0
+        query = {
+            'state': 'running',
+            'modified': {'$lt': datetime.datetime.utcnow() - datetime.timedelta(seconds=100)},
+        }
 
         while True:
+            orphan_candidate = config.db.jobs.find_one(query)
+            if orphan_candidate is None:
+                break
+
+            # If the job is currently attempting to complete, do not orphan.
+            ticket = JobTicket.find(orphan_candidate['_id'])
+            if ticket is not None and len(ticket) > 0:
```

Comment:
You could write `if ticket:` and get the same result, but it's alright if you were going for clarity.

Reply:
Yup, I really dislike random-thruthiness checks